src/Authentication/Services/SystemUserService.cs

```diff
@@ -14,133 +15,94 @@ namespace Altinn.Platform.Authentication.Services
+            _repository = systemUserRepository;
         }
 
         /// <summary>
         /// Creates a new SystemUser
         /// The unique Id for the systemuser is handled by the db.
-        /// But the calling client may send a guid for the request of creating a new system user
-        /// to ensure that there is no mismatch if the same partyId creates several new SystemUsers at the same time
         /// </summary>
-        /// <returns></returns>
-        public Task<SystemUser> CreateSystemUser(SystemUserRequestDto request, int partyId)
+        /// <returns>The SystemUser created</returns>
+        public async Task<SystemUser> CreateSystemUser(SystemUserRequestDto request, int partyId)
         {
             SystemUser newSystemUser = new()
-            {
-                Id = Guid.NewGuid().ToString(),
+            {                
                 IntegrationTitle = request.IntegrationTitle,
                 ProductName = request.ProductName,
                 OwnedByPartyId = partyId.ToString()
             };
-            theMockList.Add(newSystemUser);
-            return Task.FromResult(newSystemUser);
+
+            Guid insertedId = await _repository.InsertSystemUser(newSystemUser);            
+            SystemUser inserted = await _repository.GetSystemUserById(insertedId);
+            return inserted;
         }
 
         /// <summary>
-        /// Returns the list of SystemUsers this PartyID has registered, including "deleted" ones.
+        /// Returns the list of SystemUsers this PartyID has registered.
         /// </summary>
         /// <returns></returns>
-        public Task<List<SystemUser>> GetListOfSystemUsersPartyHas(int partyId)
+        public async Task<List<SystemUser>> GetListOfSystemUsersPartyHas(int partyId)
         {
             if (partyId < 1)
             {
-                return Task.FromResult<List<SystemUser>>(null);
+                return new List<SystemUser>(null);
             }
 
-            return Task.FromResult(theMockList);
+            return await _repository.GetAllActiveSystemUsersForParty(partyId);
         }
 
         /// <summary>
         /// Return a single SystemUser by PartyId and SystemUserId
         /// </summary>
         /// <returns></returns>
-        public Task<SystemUser> GetSingleSystemUserById(Guid systemUserId)
+        public async Task<SystemUser> GetSingleSystemUserById(Guid systemUserId)
         {
-            SystemUser search = theMockList.Find(s => s.Id == systemUserId.ToString());
+            SystemUser search = await _repository.GetSystemUserById(systemUserId);
             
-            return Task.FromResult(search);
+            return search;
         }
 
         /// <summary>
         /// Set the Delete flag on the identified SystemUser
         /// </summary>
         /// <returns></returns>
-        public Task<int> SetDeleteFlagOnSystemUser(Guid systemUserId)
+        public async Task<int> SetDeleteFlagOnSystemUser(Guid systemUserId)
```

Comment:
i guess this is a repeat of the other PR as well

Reply:
yes